src/libtiled/mapobject.cpp

```diff
@@ -428,6 +428,9 @@ void MapObject::syncWithTemplate()
 
     if (!propertyChanged(MapObject::VisibleProperty))
         setVisible(base->isVisible());
+
+    // maybe component properties can be detachable (long future)
+    mergeComponents(base->components());
```

Comment:
Hmm, I don't think this will work. Each time `syncWithTemplate` is called, we essentially copy all the components from the template? What about when the user has changed one of those properties on the instance? They'll be overwritten on loading the file, as well as when a template is edited.

Note how the properties are not copied from the template. If a property is set on an instance, it means it overrides a potential value inherited from the template. I think it could work similarly for components. Of course, the problem is that when adding a component, we'd need to add an empty map instead of one filled with the properties of the component. The Properties view would still know which properties to display, because it can look that up from the component definition, similar to how it displays properties based on the object type.

Reply:
I thought object template properties would overwrite instances'.